### DISCUSSION TITLE
가게 정보 화면 유저 닉네임 관련 오류 개선 질문입니다.

### DISCUSSION THREAD

**Discussion by labydin** [SwiftBreakers/TTEOPPOKKI4U#90] [Q&A]
StoreVC의 tableview 내부에 작성한 리뷰의 닉네임을 노출하려고 합니다. 이 때 불러온 유저 정보(userInfo)는 테이블뷰 뿐만 아니라 상세 리뷰 보기에서도 보여줍니다. 그런데 보여지는 유저 정보가 리뷰의 실제 작성자와 일치하지 않습니다. 스크롤할 때마다 값이 바껴서 함수 prepareForReuse를 썼는데 소용이 없었습니다. 혹시 비슷한 문제 겪으신 분 있으시면 도와주세요ㅠㅠ
![Simulator Screenshot - iPhone 15 Pro - 2024-06-23 at 17 20 00](https://github.com/SwiftBreakers/TTEOPPOKKI4U/assets/142779268/8bb0b8eb-5294-428f-8483-3df1fc7253c9)

![Simulator Screenshot - iPhone 15 Pro - 2024-06-23 at 17 20 35](https://github.com/SwiftBreakers/TTEOPPOKKI4U/assets/142779268/359e4c5f-2158-46f5-bf46-a71847ed00f2)
2779268/0a3f52e9-f3c1-49f1-9ac0-84ce830475b6)

**Comment by Haroldfromk** [comment]
오 흥미로운 내용이네요... 혹시 슬랙으로 현재까지 준영님이 하신 파일이나 아니면 해당부분의 코드를 좀 보여주실수 있으실까여?

안그래도 저도 이부분을 좀 생각을 했는데, 현재 계정 테스트로 인하여, 탈퇴가 반복되면서 uid가 바뀐 계정이 있습니다.
아마 제가 탈퇴 테스트를하면서 제것만 그럴가능성이 높은데, 그래서 글을 작성할때 uid를 담고 있지만, 해당 모델에 그당시의 유져의 프로필 닉네임도 같이 담아야할것같은 생각이 좀 들긴했어요, 물론 uid를 통해 realtime database에 접근하여 그 유져의 정보를 가져올 수는 있지만, 해당내용 호출이 뭔가 무의미? 하다는 생각이 들더라구요. 애초에 글정보에 uid + userNickname도 같이 있었으면 어땠을까? 라는 생각이 어제 들긴했어요

**Reply by labydin** [reply]
아 맞아요 uid: String 대신 UserModel 전체가 들어가는건 데이터 낭비지만 닉네임이나 프로필 사진 같은 건 같이 들어갔어야 했나? 싶기도 합니다. 근데 초기 모델에는 두가지 정보 기재할 생각이 없었으니까 어쩔 수 없는 부분이었다고 생각하려구요 ㅎㅎ

**Reply by Haroldfromk** [reply]
나~중에 더 고도화할때 바꾸면 되니까 지금은 이대로 가는것도 괜찮을 것 같습니다 🤣🤣
